pymc_marketing/model_builder.py

```diff
@@ -291,11 +292,8 @@ def set_idata_attrs(self, idata=None):
         if idata is None:
             raise RuntimeError("No idata provided to set attrs on.")
 
-        def default(x):
-            if isinstance(x, Prior):
-                return x.to_json()
-
-            return x.__dict__
+        def default(x) -> dict[str, Any]:
+            return x.to_json() if isinstance(x, Prior) else x.__dict__
```

Comment:
@wd60622 or I think I just need to add the case here right?

Reply:
Think so